### PR TITLE
UI settings feature

### PR DESCRIPTION
**Pull request by Aravind-psiog**
- [ ] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

**Review by Prasy12 — APPROVED** [review]

**Comment by AlejandroEsquivel** [comment]
@Prasy12 can you also let me know which node version is being used for development of the UI? this shouldn't block this PR but I will ask that we eventually add an `.nvmrc` file in order to pin the node version

File: covalent_ui/app.py
Comment (AlejandroEsquivel) [inline covalent_ui/app.py:110]: @Prasy12 is there a reason why we're enabling debug on uvicorn if we enable sdk logging?

File: covalent_ui/webapp/src/components/settings/SettingsCard.js
Comment (AlejandroEsquivel) [inline covalent_ui/webapp/src/components/settings/SettingsCard.js:180]: @Prasy12 there's an issue with this logic treating for example: `threads_per_worker` would end up having a formattedName of `Threads per_worker`, also there is code duplication of this doing the same thing in `getSubmenuName`.

I would suggest the following code is used to handle that case:

```
const formatUnderscoreConcatenatedString = (str) => {
    return str.split('_').map(word => word.charAt(0).toUpperCase() + word.slice(1)).join(" ")
  }
```

And this be re-used in `getLabelName` and `getSubmenuName`.

Also I would ask that variable naming be improved as `a` and `b` aren't very verbose.

File: covalent_ui/webapp/src/components/settings/SettingsCard.js
Comment (AlejandroEsquivel) [inline covalent_ui/webapp/src/components/settings/SettingsCard.js:166]: For any string literals used let's create a seperate file that define these as constants grouped by their usage.

e.g

```
const CONFIG_SECTIONS = {
  SLURM: 'slurm'
}
```

such that this can be re-used as `CONFIG_SECTIONS.SLURM` and similarly for the other config sections

**Review by AlejandroEsquivel — CHANGES_REQUESTED** [review]
Looks good @Prasy12 just some minor changes needed

**Comment by AlejandroEsquivel** [comment]
@Prasy12 let's also add an `.nvmrc` file that pins the node version to the one being used for development, thanks

**Comment by Prasy12** [comment]
@AlejandroEsquivel We have rectified all the points that you have mentioned above and updated the changes as well. Please review and let us know if this works good. We can merge this once you review. Thanks.

File: covalent_ui/app.py
Comment (Prasy12) [reply covalent_ui/app.py]: Fixed

**Review by AlejandroEsquivel — APPROVED** [review]

**Pull request merged**
